src/server/protocol.ts

```diff
@@ -1951,6 +1958,8 @@ namespace ts.server.protocol {
      */
     export interface QuickInfoRequest extends FileLocationRequest {
         command: CommandTypes.Quickinfo;
+        /** if true - return response with documentation as display parts instead of string */
+        richResponse?: boolean;

```

Comment:
Shouldn't this be a part of a `QuickInfoRequestArgs`?

Reply:
Same question for the completion and signature help commands. Everywhere else we only ever send `{ commandName, commandArguments }`.

Reply:
Makes sense -- there *is* no QuickInfoRequestArgs right now, which is why I missed this.

Reply:
From a usability perspective 🙂
From a figuring out what's going on by glancing at the code perspective ☹

Reply:
The protocol changes look good!